### PR TITLE
Update template for Azure ML workspace creation

### PR DESCRIPTION
**Pull request by guanyu-240**
# Best Practice Checklist

Check these items before submitting a PR... See the Contribution Guide for the full detail: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/README.md 

1. uri's compatible with all clouds (Stack, China, Government)
1. Staged artifacts use _artifactsLocation & _artifactsLocationSasToken
1. Use a parameter for resource locations with the defaultValue set to resourceGroup().location
1. Folder names for artifacts (nestedtemplates, scripts, DSC)
1. Use literal values for apiVersion (no variables)
1. Parameter files (GEN-UNIQUE for value generation and no "changemeplease" values)
1. $schema and other uris use https
1. Use uniqueString() whenever possible to generate names for resources.  While this is not required, it's one of the most common failure points in a deployment.
1. Update the metadata.json with the current date

For details: https://github.com/Azure/azure-quickstart-templates/blob/master/1-CONTRIBUTION-GUIDE/best-practices.md

- [x] - Please check this box once you've submitted the PR if you've read through the Contribution Guide and best practices checklist.

## Changelog

* Added AdbWorkspace property
* Remove Container registry provisioning since we only support ACR brought by customer
* Update the config for existing subnet to ensure it won't block private endpoint creation

### CODE REVIEW

File: 101-machine-learning-create/azuredeploy.json [hunk 1]
```diff
@@ -56,7 +56,7 @@
       },
       "storageAccountName": {
         "type": "string",
-        "defaultValue": "[concat('sa',uniqueString(resourceGroup().id))]"
+        "defaultValue": "[concat('sa',uniqueString(resourceGroup().id, deployment().name))]"
```
Comment (bmoore-msft) [inline 101-machine-learning-create/azuredeploy.json:59]: these won't be idempotent - what's the scenario you ran into for needing a change here?

Comment (guanyu-240) [reply 101-machine-learning-create/azuredeploy.json:59]: In order to ensure the uniqueness when using different deployment names.

Comment (guanyu-240) [reply 101-machine-learning-create/azuredeploy.json:59]: Without this change, if we deploy the second time(without deployment name parameter), the resource names are the same at sometime.

Comment (bmoore-msft) [reply 101-machine-learning-create/azuredeploy.json:59]: That's the intent - when deploying to the same RG you can redeploy idempotently.  If I want a new instance, I can change the param value, if I don't, this approach makes it difficult to do (and goes against the patterns in the repo.

So let's remove that from the seed

Comment (guanyu-240) [reply 101-machine-learning-create/azuredeploy.json:59]: Got it.

**Pull request merged**